### DISCUSSION TITLE
How to call refresh function in jsTree after call ajax?

### DISCUSSION THREAD

**Discussion by luantm16110560** [vakata/jstree#2554] [Q&A]
When I call refresh() function in jstree after ajax , I can not understand why it run refresh function continuously while I just call it one time, here is my code:

     $("#jstree_panel").on("open_node.jstree", function(e, data) {
    
        var url = "/api/product_bom/" + data.node.id
        $.ajax({
            type: 'GET',
            url: url,
            dataType: "json",
            success: function(resultData) {
                var bom_tree = JSON.parse(localStorage.getItem('product_bom'))
                var res = bom_tree.concat(resultData);
    
                $('#jstree_panel').jstree(true).settings.core.data = res
    
                $('#jstree_panel').jstree("refresh");
                console.log('Refreshed')
    
            },
    
        });
    
    
    });

**Comment by vakata** [comment]
refresh will also reopen any opened nodes, which will trigger the open_node event, which will trigger your ajax call, which will trigger refresh - endless loop.

**Reply by omrikaufman** [reply]
will it trigger also the selected node event ?
